core/src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala

```diff
@@ -143,23 +146,98 @@ sealed abstract class ChainParams {
+    * @return
+    */
+  def powLimit: BigInteger
+
+  /** The targetted timespan between difficulty adjustments
+    * As of this implementation, all of these are the same in bitcoin core
+    *
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L73 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L190 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L285 regtest]]
+    * */
+  def powTargetTimeSpan: Duration
+
+  /**
+    * Not sure what this value is right now, but it is constant across bitcoin
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L74 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L191 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L286 regtest]]
+    * @return
+    */
+  def powTargetSpacing: Int
+
+  /** In bitcoin [[MainNetChainParams mainnet]], the network recalculates the difficulty for the network every 2016 blocks
+    * [[https://github.com/bitcoin/bitcoin/blob/eb7daf4d600eeb631427c018a984a77a34aca66e/src/consensus/params.h#L75 bitcoin core implementation]]
+    * */
+  def difficultyChangeInterval: Long = {
+    powTargetTimeSpan.toSeconds / powTargetSpacing
+  }
 }
 
-sealed abstract class BitcoinChainParams extends ChainParams
+sealed abstract class BitcoinChainParams extends ChainParams {
+
+  /** The targetted timespan between difficulty adjustments
+    * As of this implementation, all of these are the same in bitcoin core
+    *
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L73 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L190 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L285 regtest]]
+    * */
+  override lazy val powTargetTimeSpan: Duration = {
+    val time = 14 * 24 * 60 * 60 //two weeks
+    time.seconds
+  }
+
+  /**
+    * Not sure what this value is right now, but it is constant across bitcoin
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L74 mainnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L191 testnet]]
+    * [[https://github.com/bitcoin/bitcoin/blob/a083f75ba79d465f15fddba7b00ca02e31bb3d40/src/chainparams.cpp#L286 regtest]]
+    * @return
+    */
+  override lazy val powTargetSpacing: Int = {
+    10 * 60 //what are these magic numbers?
+  }
+
+  /** The best chain should have this amount of work */
```

Comment:
Any reason for this to be a `java.math.BigInteger` and not `scala.math.BigInt`? BigInt has implicit conversions from Int

Reply:
Not really, just thought it would be more friendly to java people.